api/utils/extapi.go

```diff
@@ -0,0 +1,35 @@
+package utils
+
+import (
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+func FetchInfo(idParam int) string {
+
+	id := strconv.Itoa(idParam)
+
+	resp, err := http.Get("https://rupaulsdragrace.fandom.com/api.php/?action=query&prop=extracts&exlimit=1&explaintext=true&pageids=" + id + "&format=json")
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	plainBody := string(body)
+	object := strings.Split(plainBody, "{")[4]
+	extract := strings.Split(object, "\"extract\":")[1]
+	bio := strings.Split(extract, "==")[0]
+	trimmedBio := strings.Replace(
+		strings.Replace(
+			strings.Replace(bio, "\"", "", -1), "\\n", "", -1), ",", "", -1)
```

Comment:
The reason I'm parsing the response as plain text is because the JSON response from this API is not standard. The names of the objects vary depending on the page ID, and Go does not allow for variable names in structs. 

This is why I asked this during the second or third workshop, so I figured this would be the best workaround.

See sample response from API by calling enpoint:
https://rupaulsdragrace.fandom.com/api.php/?action=query&prop=extracts&exlimit=1&explaintext=true&titles=Paper_Cut&format=json 

Notice how the object `pages{}` has individual objects with a unique ID instead of an array of standardized objects.

Reply:
You can actually unmarshal this type of dynamic responses as a map and you would end up with something like this and you wouldn't have to do manual extraction of the fields:

```
type Response struct {
  BatchComplete string `json:"batchComplete"` 
  Query QueryResp `json:"query"`
  .
  .
  .
}

type QueryResp struct {
  Pages map[int]Page `json:"pages"`
}

type Page struct {
  PageId int `json:"pageid"`
  .
  .
  .
}
```

I would suggest looking at this answer for more insight: https://stackoverflow.com/questions/29347092/how-to-parse-deserialize-dynamic-json